From faa605dca56c7474fb60b634f1f22732d3cf7dac Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Tue, 13 Nov 2018 11:36:29 +0200
Subject: [PATCH 5114/5242] MLK-20189-8: ASoC: fsl_sai: use signed offset
 variable

commit  7b73200fa6bea1891e968077fb02f4cd9a456116 from
https://source.codeaurora.org/external/imx/linux-imx.git

Both dataline_off and dataline_off_dsd fields are unsigned,
thus checking negative values make no sense. Use a signed
variable to calculate offset instead.

This fixes Coverity issue: CID1899299

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_sai.c |   14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index c464cbc8..ff1a5bad 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -1262,7 +1262,7 @@ static int fsl_sai_probe(struct platform_device *pdev)
 	char tmp[8];
 	int irq, ret, i;
 	int index;
-	int firstbitidx, nextbitidx;
+	int firstbitidx, nextbitidx, offset;
 	u32 buffer_size;
 	struct regmap_config fsl_sai_regmap_config = fsl_sai_v2_regmap_config;
 	unsigned long irqflags = 0;
@@ -1346,10 +1346,8 @@ static int fsl_sai_probe(struct platform_device *pdev)
 	for (i = 0; i < 2; i++) {
 		firstbitidx = find_first_bit((const unsigned long *)&sai->dataline[i], 8);
 		nextbitidx = find_next_bit((const unsigned long *)&sai->dataline[i], 8, firstbitidx+1);
-		sai->dataline_off[i] = nextbitidx - firstbitidx - 1;
-
-		if (sai->dataline_off[i] < 0 || sai->dataline_off[i] >= 7)
-			sai->dataline_off[i] = 0;
+		offset = nextbitidx - firstbitidx - 1;
+		sai->dataline_off[i] = (offset < 0 || offset >= 7 ? 0 : offset);
 	}
 
 	ret = of_property_read_u32_index(np, "fsl,dataline,dsd", 0, &sai->dataline_dsd[0]);
@@ -1368,10 +1366,8 @@ static int fsl_sai_probe(struct platform_device *pdev)
 	for (i = 0; i < 2; i++) {
 		firstbitidx = find_first_bit((const unsigned long *)&sai->dataline_dsd[i], 8);
 		nextbitidx = find_next_bit((const unsigned long *)&sai->dataline_dsd[i], 8, firstbitidx+1);
-		sai->dataline_off_dsd[i] = nextbitidx - firstbitidx - 1;
-
-		if (sai->dataline_off_dsd[i] < 0 || sai->dataline_off_dsd[i] >= 7)
-			sai->dataline_off_dsd[i] = 0;
+		offset = nextbitidx - firstbitidx - 1;
+		sai->dataline_off_dsd[i] = (offset < 0 || offset >= 7 ? 0 : offset);
 	}
 
 	if ((of_find_property(np, "fsl,i2s-xtor", NULL) != NULL) ||
-- 
1.7.9.5

